From 06c4e2305c179a2f445f45e94279c733f505d528 Mon Sep 17 00:00:00 2001
From: Montel Laurent <montel@kde.org>
Date: Sat, 29 Mar 2014 16:56:56 +0100
Subject: [PATCH 15/48] Normalize email address before to store it in baloo

=> better to save quote email during saving when necessary

CCMAIL: faure@kde.org
---
 src/pim/agent/CMakeLists.txt                   |  1 +
 src/pim/agent/autotests/CMakeLists.txt         |  3 ++-
 src/pim/agent/emailindexer.cpp                 | 18 +++++-------------
 src/pim/agent/tests/CMakeLists.txt             |  2 +-
 src/pim/akonadiplugin/autotests/CMakeLists.txt |  1 +
 5 files changed, 10 insertions(+), 15 deletions(-)

diff --git a/src/pim/agent/CMakeLists.txt b/src/pim/agent/CMakeLists.txt
index 03427b2..efdca9b 100644
--- a/src/pim/agent/CMakeLists.txt
+++ b/src/pim/agent/CMakeLists.txt
@@ -27,6 +27,7 @@ target_link_libraries(akonadi_baloo_indexer
     ${KDEPIMLIBS_KMIME_LIBS}
     ${KDEPIMLIBS_KABC_LIBS}
     ${XAPIAN_LIBRARIES}
+    ${KDEPIMLIBS_KPIMUTILS_LIBS}
 )
 
 install(TARGETS akonadi_baloo_indexer ${INSTALL_TARGETS_DEFAULT_ARGS})
diff --git a/src/pim/agent/autotests/CMakeLists.txt b/src/pim/agent/autotests/CMakeLists.txt
index a706f74..5ee987a 100644
--- a/src/pim/agent/autotests/CMakeLists.txt
+++ b/src/pim/agent/autotests/CMakeLists.txt
@@ -24,6 +24,7 @@ target_link_libraries(indexertest
     ${KDEPIMLIBS_KMIME_LIBS}
     ${KDEPIMLIBS_KABC_LIBS}
     ${XAPIAN_LIBRARIES}
+    ${KDEPIMLIBS_KPIMUTILS_LIBS}
     baloocore
     balooxapian
-)
\ No newline at end of file
+)
diff --git a/src/pim/agent/emailindexer.cpp b/src/pim/agent/emailindexer.cpp
index 989e4dd..0176c27 100644
--- a/src/pim/agent/emailindexer.cpp
+++ b/src/pim/agent/emailindexer.cpp
@@ -26,6 +26,8 @@
 #include <Akonadi/KMime/MessageFlags>
 
 #include <QTextDocument>
+#include <kpimutils/email.h>
+
 
 EmailIndexer::EmailIndexer(const QString& path, const QString& contactDbPath):
     AbstractIndexer(), m_doc( 0 ), m_termGen( 0 )
@@ -115,19 +117,9 @@ namespace {
     // TODO: Move this into KMime?
     // TODO: If name is all upper/lower then try to captialize it?
     QString prettyAddress(const KMime::Types::Mailbox& mbox) {
-        QString name = mbox.name().simplified();
-        QByteArray email = mbox.address().simplified().toLower();
-
-        // Remove outer quotes recursively
-        while (name.size() >= 2 && (name[0] == '\'' || name[0] == '"') &&
-               (name[name.size()-1] == '\'' || name[name.size()-1] == '"')) {
-            name = name.mid(1, name.size()-2);
-        }
-
-        if (name.isEmpty())
-            return email;
-        else
-            return name + QLatin1String(" <") + QString::fromUtf8(email) + QLatin1Char('>');
+        const QString name = mbox.name().simplified();
+        const QByteArray email = mbox.address().simplified().toLower();
+        return KPIMUtils::normalizedAddress(name, QString::fromUtf8(email));
     }
 }
 
diff --git a/src/pim/agent/tests/CMakeLists.txt b/src/pim/agent/tests/CMakeLists.txt
index 627623c..ca33db3 100644
--- a/src/pim/agent/tests/CMakeLists.txt
+++ b/src/pim/agent/tests/CMakeLists.txt
@@ -12,6 +12,6 @@ target_link_libraries(emailindexer
   ${KDEPIMLIBS_AKONADI_LIBS}
   ${KDEPIMLIBS_AKONADI_KMIME_LIBS}
   ${KDEPIMLIBS_KMIME_LIBS}
-
+  ${KDEPIMLIBS_KPIMUTILS_LIBS}
   ${XAPIAN_LIBRARIES}
 )
diff --git a/src/pim/akonadiplugin/autotests/CMakeLists.txt b/src/pim/akonadiplugin/autotests/CMakeLists.txt
index c4a044f..c171204 100644
--- a/src/pim/akonadiplugin/autotests/CMakeLists.txt
+++ b/src/pim/akonadiplugin/autotests/CMakeLists.txt
@@ -27,6 +27,7 @@ target_link_libraries(searchplugintest
     ${KDEPIMLIBS_KMIME_LIBS}
     ${KDEPIMLIBS_KABC_LIBS}
     ${XAPIAN_LIBRARIES}
+    ${KDEPIMLIBS_KPIMUTILS_LIBS}
     baloocore
     balooxapian
 )
-- 
1.9.1

